### PR TITLE
Hw layer rearrangement 2

### PR DESCRIPTION
**Pull request by dron0gus**

### CODE REVIEW

**Comment by dron0gus** [comment]
@mck1117 can you review? Part where I was touching mass storage. Or all PR.

File: simulator/halconf.h [hunk 1]
```diff
@@ -372,6 +372,14 @@
 #define USB_USE_WAIT                FALSE
 #endif
 
+/* halconf_community.h */
+/**
+ * @brief   Enables the USB_MSD subsystem.
+ */
+#if !defined(HAL_USE_USB_MSD) || defined(__DOXYGEN__)
+#define HAL_USE_USB_MSD             FALSE
```
Comment (mck1117) [inline simulator/halconf.h:380]: does simulator not include `rusefi_halconf.h`? Should it?

**Comment by mck1117** [comment]
sorry, I think I merge conflicted you 🙃

**Comment by dron0gus** [comment]
> sorry, I think I merge conflicted you upside_down_face

NP, I was expecting this.

File: simulator/halconf.h (hunk 1 above)
Comment (dron0gus) [reply simulator/halconf.h:380]: Ok. Will create it.

**Comment by rusefillc** [comment]
@dron0gus poke :)

**Comment by rusefillc** [comment]
@dron0gus poke?

**Comment by rusefillc** [comment]
@dron0gus close? finish? poke later?

**Comment by dron0gus** [comment]
@rusefillc too much to rebase. I'll do it someday later. For now - close.

**Pull request closed without merging**